### PR TITLE
feat(ci): run acceptance tests with kona node and op-reth

### LINKED ISSUE #2946: feat(tests/acceptance): integrate op-reth/kona acceptance-tests
## Description

We should be running the acceptance tests for the rust stack to ensure that we have enough coverage when doing hardfork updates.

- [x] Add just commands to run acceptance tests #2902 
- [x] Integrate acceptance tests in CI. #2902 https://github.com/ethereum-optimism/optimism/pull/17882

## Context

We currently testing `kona-node` and `op-reth` with the following tools:
- unit tests: for `kona-node` and `op-reth`. Don't cover network interactions or p2p incompatibilities
- action-tests: for `kona-proof`. Indirectly test `op-alloy`/`op-revm` crates. Don't test `op-reth` directly
- e2e tests with `sysgo` using the `kona`'s testing suite (https://github.com/op-rs/kona/tree/main/tests): cover network communication, some p2p interactions. Don't systematically cover hardfork specific behavior/edge cases. Test both `kona-node` and `op-reth` under several network configurations
- devnets (netchef): mostly for `op-reth`. Don't run networks with `kona-node` yet

What we're currently missing is to run the acceptance tests suite for `op-reth` and `kona-node` to ensure we have adequate coverage for edge case features and hardfork specific behavior. The `base` acceptance test suite (https://github.com/ethereum-optimism/optimism/tree/develop/op-acceptance-tests/tests/base) is currently failing for the `kona-node`/`op-reth` and `op-node`/`op-reth` couples. Specific tests failing are: 
- https://github.com/ethereum-optimism/optimism/tree/develop/op-acceptance-tests/tests/fjord - `TestFees`
- https://github.com/ethereum-optimism/optimism/tree/develop/op-acceptance-tests/tests/isthmus/operator_fee - `NonZeroFees`
- https://github.com/ethereum-optimism/optimism/tree/develop/op-acceptance-tests/tests/isthmus/withdrawal_root - `TestWithdrawalRoot`

The acceptance tests pass for the `op-node/op-geth` couples

### PR DESCRIPTION
**Pull request by theochap**
## Description

Adds a ci job to run acceptance tests for kona-node and op-reth in CI

Close #2946

### CODE REVIEW

File: .github/workflows/node_e2e_sysgo_tests.yaml [hunk 1]
```diff
@@ -37,11 +37,11 @@ jobs:
 
       - uses: jdx/mise-action@v3       # installs Mise + runs `mise install`
 
-      - name: Setup Go 1.24.3
+      - name: Setup Go 1.23.8
```
Comment (emhane) [inline .github/workflows/node_e2e_sysgo_tests.yaml:40]: is this intentional?
